src/fingerprinting/utils.ts

```diff
@@ -16,6 +16,25 @@ export const getGeneratorDefaultOptions = (launchContext: LaunchContext): Finger
     return options;
 };
 
+export const mergeArgsToHideWebdriver = (originalArgs: string[]): string[] => {
+    if (originalArgs && originalArgs.length) {
+        const argumentIndex = originalArgs.findIndex((arg: string) => arg.startsWith('--disable-blink-features='));
+        const hasArgument = argumentIndex !== -1;
+
+        if (hasArgument) {
+            const arg = originalArgs[argumentIndex];
+            // Append our argument.
+            originalArgs[argumentIndex] = `${arg},AutomationControlled`;
+        } else {
+            originalArgs.push('--disable-blink-features=AutomationControlled');
+        }
+
+        return originalArgs;
+    }
+
+    return ['--disable-blink-features=AutomationControlled'];
```

Comment:
this can be greatly simplified
- if you swap the branches, you have less nesting
- you can use optional chaining
- use `+=` to append to the already defined argument
- I would also inline the `hasArgument`, as it seems to be trivial to understand (maybe using `argumentIndex >= 0` and swapping branches would be even more readable)

```suggestion
    if (!originalArgs?.length) {
      return ['--disable-blink-features=AutomationControlled'];
    }

    const argumentIndex = originalArgs.findIndex((arg: string) => arg.startsWith('--disable-blink-features='));

    if (argumentIndex !== -1) {
      originalArgs[argumentIndex] += ',AutomationControlled';
    } else {
      originalArgs.push('--disable-blink-features=AutomationControlled');
    }

    return originalArgs;
```